### PR TITLE
Add new metrics for handling stream error

### PR DESCRIPTION
**Pull request by xwduan**
## What changed?

1. Add metrics to monitor non stream error on replication stack
2. Unify error log prefix, so when ^ alarm fired, we can easily search log by prefix
## Why?

To better handling stream error and replication error
## How did you test it?

n/a
## Potential risks

n/a
## Documentation

n/a

### CODE REVIEW

File: service/history/replication/stream_receiver.go [hunk 1]
```diff
@@ -185,7 +189,11 @@ func (r *StreamReceiverImpl) recvEventLoop() error {
 
 	err := r.processMessages(r.stream)
 
-	r.logger.Error("StreamReceiver exit recv loop", tag.Error(err))
+	if streamError, ok := err.(*StreamError); ok {
```
Comment (tdeebswihart) [inline service/history/replication/stream_receiver.go:192]: Please use `errors.As` instead

File: service/history/replication/stream.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func WrapEventLoop(
 ) {
 	defer streamStopper()
 
-	for retryCount := 0; retryCount < 11; retryCount++ {
+	for {
```
Comment (tdeebswihart) [inline service/history/replication/stream.go:74]: Is this change intentional?

Comment (xwduan) [reply service/history/replication/stream.go:74]: Yes it is. This change is to not disconnect the grpc stream if it is `ReplicationServiceError`. Initially it is infinite loop here. I added the retry counts at previous version just in case we miss categorize the error. But we learned from last Friday's incident that retry count here will still disconnect the stream and cause issue on network layer.

So I added additional metrics and planing to add alarm on them, so in case we miss category the error, we still get notified.

File: service/history/replication/stream_receiver.go (hunk 1 above)
Comment (yux0) [inline service/history/replication/stream_receiver.go:192]: Maybe abstract to a util function and use in both places.

File: service/history/replication/stream_sender.go [hunk 3]
```diff
@@ -156,13 +156,13 @@ func (s *StreamSenderImpl) recvEventLoop() (retErr error) {
 	for !s.shutdownChan.IsShutdown() {
 		req, err := s.server.Recv()
 		if err != nil {
-			s.logger.Error("StreamSender exit recv loop", tag.Error(err))
-			return NewStreamError("Stream recv error", err)
+			s.logger.Error("GrpcStreamError StreamSender failed to receive", tag.Error(err))
```
Comment (yux0) [inline service/history/replication/stream_sender.go:159]: ReplicationStreamError?

**Review by yux0 — APPROVED** [review]

**Review by meiliang86 — APPROVED** [review]
